tests/psbt.rs

```diff
@@ -0,0 +1,704 @@
+//! This file implements the test vector workflow in:
+//!
+//!  https://github.com/bitcoin/bips/blob/master/bip-0174.mediawiki#test-vectors
+//!
+//! starting at the line:
+//!
+//!  The private keys in the tests below are derived from the following master private key:
+//!
+
+// Attribution: `sign_psbt` logic is based on code in the `wallet` module of the `bdk` project. The
+// code in the two functions `legacy_sighash()` and `segwit_v0_sighash` is copied directly from
+// `bdk`, hence their licence applies to that code.
+//
+// ref: http://github.com/bitcoindevkit/bdk
+//
+// Bitcoin Dev Kit
+// Written in 2020 by Alekos Filini <alekos.filini@gmail.com>
+//
+// Copyright (c) 2020-2021 Bitcoin Dev Kit Developers
+//
+// This file is licensed under the Apache License, Version 2.0 <LICENSE-APACHE
+// or http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your option.
+// You may not use this file except in accordance with one or both of these
+// licenses.
```

Comment:
I think we should ask the BDK people to allow re-license the code here to be compatible(If they are different) with what rust-bitcoin currently has. 

I think we can remove the license here(while retaining the credits) if @afilini allows it.

Reply:
We have three open PRs that all use this code (#940, #946, and this one). Lets defer the licensing stuff to #946 and apply whatever decision is made there to the other two. I don't think we should merge any of them until this is resolved.

Reply:
See my answer here: https://github.com/bitcoindevkit/bdk/issues/586#issuecomment-1102266447

Reply:
`sign_psbt` code remains a draft because of taproot. We _could_ move forward as-is, including the BDK license & attribution in this file, based on the discussion in https://github.com/bitcoindevkit/bdk/issues/586#issuecomment-1102266447

Reply:
If you want to push this in before #957 you could pull the ecdsa signing code out of 957 in place of what we did here, and remove the attribution stuff. Saves reviewers bothering with license issues. 

Although, in my opinion, it would be best to rebase this on top of 957 and leave it as draft until 957 goes in but explicitly stating in the PR description that this PR validates/tests the signing code in #957. We can then do the same with #940 and #999. Also this would mean that you, myself, and @dunxen all have a vested interest in getting 957 correct and merged :)

FTR I have previously rebased this work on top of 957 and it all works.